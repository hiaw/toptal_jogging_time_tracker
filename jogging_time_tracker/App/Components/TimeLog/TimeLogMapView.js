/* @flow */
import React from 'react'
import MapView from 'react-native-maps'

const styles = {
  map: {
    height: 400,
    width: 400,
  },
}

const TimeLogMapView = (props: NewProps) => {
  const { coordinate } = props

  const latitudeDelta = 0.1
  const longitudeDelta = 0.1

  if (coordinate && coordinate.latitude && coordinate.longitude) {
    const { latitude, longitude } = coordinate
    return (
      <MapView
        style={styles.map}
        initialRegion={{
          latitude,
          longitude,
          latitudeDelta,
          longitudeDelta,
        }}
      >
        <MapView.Marker coordinate={coordinate} />
      </MapView>
    )
  }

  return null
}

export default TimeLogMapView
