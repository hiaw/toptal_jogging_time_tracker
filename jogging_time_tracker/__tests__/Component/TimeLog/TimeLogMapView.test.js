import 'react-native'
import React from 'react'

import TimeLogMapView from '../../../App/Components/TimeLog/TimeLogMapView.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('react-native-maps')

it('renders without coordiante', () => {
  const tree = renderer.create(<TimeLogMapView />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders with coordiante', () => {
  const tree = renderer
    .create(<TimeLogMapView coordinate={{ latitude: 12, longitude: 12 }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
