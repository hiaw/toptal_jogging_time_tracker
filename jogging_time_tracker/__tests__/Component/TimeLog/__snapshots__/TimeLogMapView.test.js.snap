// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with coordiante 1`] = `null`;

exports[`renders without coordiante 1`] = `null`;
